### DISCUSSION TITLE
我觉得deepin&UOS下的应用不应该永久性的阻止电脑关机

### DISCUSSION THREAD

**Discussion by widon1104** [linuxdeepin/developer-center#3642] [Features Request & Ideas | 特性请求 & 头脑风暴]
现在deepin&UOS下面，如果qq未关闭，这时候关机是无法成功关机的，会弹出来框提示是否关闭qq继续关机。
如果这时候什么都不点击，就会一直hang在这里，几个小时，几天都这样了，我觉得这个不太合理，
如果提示框出来有应用阻止关机，**我觉得这时候如果用户5分钟没有操作，就应该把应用杀掉，继续关机**，
我觉得这样处理，比一直hang住要好的多。因为关机可能是shutdown -h触发的定时关机，可能是更新几个小时再触发的关机动作，等待到关机的时候，可能用户并不在电脑前面了，这时候一直hang住是不对的。

**Comment by BLumia** [comment]
> 如果用户5分钟没有操作，就应该把应用杀掉，继续关机

如果用户不知道这个“五分钟”的机制，那么相当于操作系统在做用户意料之外的事情，并且意味着应用软件未能成功避免用户潜在的可能造成文件损坏的行为。

相比而言，在界面提供一个强制关机，由用户显式的表示“理解风险，就是要继续关机”反而更合理。
